### ISSUE TITLE
FNX2-13397 ⁃ Investigate MAIN startup performance on low-end devices (G5 preferred)

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#11244]
Let's take a profile and find any low-hanging fruit. File follow-ups at the top of the backlog for any issues found.

We'd prefer a simpleperf profile but, if that's not possible, the Android Profiler will do.

**Comment by sraturi** [comment]
**Profile**: [Dowload and open in browser](https://drive.google.com/file/d/1WPx_CVS-uvH4RJTn0eSNq9NOFNWkY86L/view?usp=sharing)

  
  

**TLDR**: Didn't find anything significant on the profile but saw The `MigratingFenixApplication/Activity` take quite a bit time every time we do a cold startup.

  

Things noted:

1.  setupTheme() - > takes around 40-60 milliseconds to access sharedpref
    
2.  setupCrashReporting() -> can be moved to an IO coroutine

**FNPRMS with aurora build (2020.07.07) on s7**

1.  Moved setup setupTheme() and setupCrashReporting() to IO coroutine

 

Master | with `setupTheme() `in IO coroutine | With `setupCrashReporting()` in IO coroutine | With `setupCrashReporting()` and `setupTheme()` in IO coroutine
-- | -- | -- | --
1.0572692307692309 | 1.0413478260869564 | 1.0302592592592592 | 1.0052352941176472


**Note:**
The master seem to land between **1.02 - 1.08..** and the changes(last column) went as low as **0.99** seconds

**Comment by mcomella** [comment]
Closing: we're revisiting startup holistically.
